src/java/io/bazel/rulesscala/scalac/ScalaCInvoker.java

```diff
@@ -255,6 +255,37 @@ private static void processRequest(List<String> args) throws Exception {
           reporter.flush();
           throw new RuntimeException("Build failed");
       } else {
+        /**
+         * See if there are java sources to compile
+         */
+        if (ops.javaFiles.length > 0) {
+          StringBuilder cmd = new StringBuilder();
+          cmd.append(ops.javacPath);
+          if (ops.jvmFlags != "") cmd.append(ops.jvmFlags);
```

Comment:
If you do that, it passes empty strings to the javac which then errors
rather than ignoring empty args.

It's great. I'm proud of this. ;)
On Mon, Sep 12, 2016 at 14:39 ianoc <notifications@github.com> wrote:

> In src/java/io/bazel/rulesscala/scalac/ScalaCInvoker.java
> <https://github.com/ianoc/rules_scala/pull/2#discussion_r78479922>:
>
> > @@ -255,6 +255,37 @@ private static void processRequest(List<String> args) throws Exception {
> >            reporter.flush();
> >            throw new RuntimeException("Build failed");
> >        } else {
> > +        /**
> > +         * See if there are java sources to compile
> > +         */
> > +        if (ops.javaFiles.length > 0) {
> > +          StringBuilder cmd = new StringBuilder();
> > +          cmd.append(ops.javacPath);
> > +          if (ops.jvmFlags != "") cmd.append(ops.jvmFlags);
>
> should these not be separate args to the Process builder rather than a
> string concat?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ianoc/rules_scala/pull/2/files/d0d6658d1f6ff948949ac47ea1ab7f9f128c8606#r78479922>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEJduvOcPE3K8mTZEnbhLDkvK0h8Yvkks5qpfDCgaJpZM4J7LKQ>
> .
>